File: coin_0.c
#include <stdio.h>

int main(void) {
    int N, K, i;
    int cnt = 0;
    int arr[10];
    scanf("%d %d", &N, &K);
    for( i=0; i<N; i++){
        scanf("%d", &arr[i]);
    }
    i = N-1;
    while(K != 0){
        if(K >= arr[i]){
            K=K-arr[i];
            cnt++;
        }
        else {
            i--;
        }
    }
    printf("필요 동전 개수의 최솟값 : %d", cnt);
}

File: coin_0.java
import java.util.Scanner;
 
public class Main {
 
	public static void main(String[] args) {
 
		Scanner in = new Scanner(System.in);
		
		int N = in.nextInt();
		int K = in.nextInt();
		int i;
		int cnt = 0;
		int[] arr = new int[N];
		
		for(i = 0; i < N; i++) {
			arr[i] = in.nextInt();
		}
		
		for(i = N - 1; i >= 0; i--) {
 
			if(arr[i] <= K) {
				cnt += (K / arr[i]);
				K = K % arr[i];
			}
		}
		System.out.println("필요 동전 개수의 최솟값 : " + cnt);
	}
 
}